src/main/java/io/split/android/client/SyncConfig.java

```diff
@@ -0,0 +1,91 @@
+package io.split.android.client;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.split.android.client.utils.Logger;
+import io.split.android.client.validators.SplitValidator;
+import io.split.android.client.validators.SplitValidatorImpl;
+
+import static androidx.core.util.Preconditions.checkNotNull;
+
+class SyncConfig {
+
+    private final static int MAX_BY_NAME_VALUES = 400;
+    private final static int MAX_BY_PREFIX_VALUES = 50;
+    private final List<SplitFilter> mFilters;
+
+
+    private SyncConfig(List<SplitFilter> filters) {
+        mFilters = filters;
+    }
+
+    public List<SplitFilter> getFilters() {
+        return mFilters;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private List<SplitFilter> mBuilderFilters = new ArrayList<>();
+        private final SplitValidator mSplitValidator = new SplitValidatorImpl();
+
+        public SyncConfig build() {
+            List<SplitFilter> filters = new ArrayList<>();
+            for (SplitFilter filter : mBuilderFilters) {
+
+                List<String> values = filter.getValues();
+                switch (filter.getType()) {
+                    case BY_NAME:
+                        if (values.size() > MAX_BY_NAME_VALUES) {
```

Comment:
You should move the limit validation after invalid and duplicated values are removed, and do it only once when all values are grouped by type.